docs/src/main/asciidoc/infinispan-client-reference.adoc

```diff
@@ -795,8 +850,37 @@ public class Book {
 You can use either the Query DSL or the Ickle Query language with the Quarkus Infinispan client
 extension.
 
+.Query.java
+[source,java]
+----
+@Inject
+@Remote("books")
+RemoteCache<String, Book> booksCache; //<1>
+
+Query<Book> query = booksCache.query("from book_sample.Book b where b.authors.name like '%" + name + "%'"); //<2>
+List<Book> list = query.execute().list();
+----
+<1> Inject the books cache
+<2> Perform a full text query on books author name
+
 NOTE: You can read more about https://infinispan.org/docs/stable/titles/query/query.html[querying] in the Infinispan documentation.
 
+[IMPORTANT]
+====
+Previous to Infinispan 15 integration, queries were executed by calling the following code:
+.Query.java
+[source,java]
+----
+QueryFactory queryFactory = Search.getQueryFactory(booksCache); <1>
+Query query = queryFactory.create("from book_sample.Book");
+List<Book> list = query.execute().list();
+----
+<1> Breaking change in 3.9
+
+This code won't work since ```RemoteCache``` is ```@ApplicationScoped``` proxy bean.
+This code is deprecated and will be removed. Simplify with Infinispan 15 or above
+by using the provided `query` method in the `RemoteCache` API.
```

Comment:
im not sure what is meant here - is says it wont work but also that the code is deprecated ...so does the deprecated warning happen AND the code wont work or how to undestand it?

Reply:
The code is deprecated, it will still work outside Quarkus. Inside Quarkus won't work since the RemoteCache is ApplicationScope bean with this change, and will fail on start with a classcast exception. That's the reason why we could not @InjectMock 
To make query work on Quarkus 3.9, the user must change the code from

```java
QueryFactory queryFactory = Search.getQueryFactory(peopleCache);
Query query = queryFactory.create("FROM tutorial.Person p where p.lastName = :lastName");
query.setParameter("lastName", "Andersen");

//same
List<Person> result = query.execute().list();
```

to 

```java
Query<Person> query = peopleCache.<Person>query("FROM tutorial.Person p where p.lastName = :lastName");
query.setParameter("lastName", "Andersen");

// same
List<Person> result = query.execute().list();
```

Reply:
Looks the message is unclear, I'll change it